File: RelationExample/RelationExample/Rectangle.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RelationExample
{
    class Rectangle  : Quadrangle
    {

        public Rectangle(float sideLengthA, float sideLengthB) : base(sideLengthA, sideLengthB, sideLengthA, sideLengthB)
        {}

        public float CalculateArea()
        {
            return sideLengthA * sideLengthB;
        }

        public override void PrintPerimeter(float perimeter)
        {
            Console.WriteLine("The perimeter of the rectangle is: " + perimeter);
        }

        public override void PrintArea(float area)
        {
            Console.WriteLine("The area of the rectangle is: " + area);
        }
    }
}

File: RelationExampleC++/RelationExampleC++/Rectangle.cpp
#include "Rectangle.h"

Rectangle::Rectangle(float sideLengthA, float sideLengthB) : Quadrangle(sideLengthA, sideLengthB, sideLengthA, sideLengthB)
{}

float Rectangle::CalculateArea()
{
	return m_sideLengthA * m_sideLengthB;
}

void Rectangle::PrintPerimeter(float perimeter)
{
	std::cout << "The perimeter of the rectangle is: " << perimeter << "\n";
}

void Rectangle::PrintArea(float area)
{
	std::cout << "The area of the rectangle is: " << area << "\n";
}